backend/worker/kafka_worker.go

```diff
@@ -126,8 +126,25 @@ type KafkaWorker struct {
 	WorkerThread int
 }
 
+func (k *KafkaBatchWorker) log(ctx context.Context, fields log.Fields, msg ...interface{}) {
+	if k.lastPartitionId != nil {
+		return
+	}
```

Comment:
i think this is flipped? you want to return if the lastPartitionId is nil, to avoid dereferencing it
```suggestion
	if k.lastPartitionId == nil {
		return
	}
```